client/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list.jsx

```diff
@@ -0,0 +1,173 @@
+import { Button, Gridicon } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+import PropTypes from 'prop-types';
+import { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { validateAllFields } from 'calypso/lib/domains/email-forwarding';
+import EmailForwardingAddNewCompact from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact';
+import { emailManagement } from 'calypso/my-sites/email/paths';
+import {
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'calypso/state/analytics/actions';
+import { addEmailForward } from 'calypso/state/email-forwarding/actions';
+import getEmailForwardingLimit from 'calypso/state/selectors/get-email-forwarding-limit';
+import getEmailForwardingType from 'calypso/state/selectors/get-email-forwarding-type';
+import { getEmailForwards, isAddingForward } from 'calypso/state/selectors/get-email-forwards';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+class EmailForwardingAddNewCompactList extends Component {
+	static propTypes = {
+		emailForwards: PropTypes.array,
+		emailForwardingLimit: PropTypes.number,
+		selectedDomainName: PropTypes.string.isRequired,
+	};
+
+	state = {
+		forwards: [ { destination: '', mailbox: '', valid: false } ],
+	};
+
+	hasForwards() {
+		return this.props.emailForwards.length > 0;
+	}
+
+	hasReachedLimit() {
+		return this.props.emailForwards.length >= this.props.emailForwardingLimit;
+	}
+
+	validForwards() {
+		return ! this.state.forwards.some( ( t ) => ! t.valid );
+	}
+
+	addNewEmailForwardWithAnalytics = ( domainName, mailbox, destination, siteSlug ) => {
+		const onSuccessRedirectTarget = () => {
+			page( emailManagement( siteSlug, domainName ) );
+		};
+
+		withAnalytics(
+			composeAnalytics(
+				recordTracksEvent(
+					'calypso_domain_management_email_forwarding_add_new_email_forward_click',
+					{
+						destination,
+						domain_name: domainName,
+						mailbox,
+					}
+				),
+				this.props.addEmailForward( domainName, mailbox, destination, onSuccessRedirectTarget )
+			)
+		);
+	};
+
+	addNewEmailForwardsClick = ( event ) => {
+		const { selectedSiteSlug, addingForward } = this.props;
+
+		event.preventDefault();
+
+		if ( addingForward ) {
```

Comment:
I understand why you didn't name `addingForward` as `isAddingForward` but that's something we could work around by not destructing this variable:

```suggestion
		if ( this.props.isAddingForward ) {
```